src/Microsoft.IdentityModel.Tokens.Saml/SamlTokenUtilities.cs

```diff
@@ -102,5 +104,60 @@ internal static IEnumerable<SecurityKey> GetKeysForTokenSignatureValidation(stri
             }
             return null;
         }
+
+        /// <summary>
+        /// Creates claims from  a dictionary.
+        /// </summary>
+        /// <param name="claimsCollection"> A dictionary that represents a set of claims.</param>
+        /// <returns> A collection of Claim objects created from the dictionary.</returns>
+        internal static IEnumerable<Claim> CreateClaimsFromDictionary(IDictionary<string, object> claimsCollection)
+        {
+            List<Claim> claims = null;
+            object value;
+            if (claimsCollection == null)
+                return claims;
+
+            claims = new List<Claim>();
+            foreach (string claimtype in claimsCollection.Keys)
+            {
+                if (claimsCollection.TryGetValue(claimtype, out value))
```

Comment:
The other reason for iterating over the KeyValuePair instead of the Key is that you don't need the `TryGetValue`. You have direct access to the Key and the value and end up with a lower cyclomatic complexity value / more simple code for this method.

Reply:
Thank you. Its a new learning for me. It makes sense considering cyclomatic complexity.